src/workerd/io/io-own.h

```diff
@@ -236,6 +258,35 @@ class IoPtr {
       : deleteQueue(kj::mv(deleteQueue)), ptr(ptr) {}
 };
 
+// Owned pointer held by a KJ I/O object living in the same thread as an IoContext. The underlying
+// object is destroyed when the ReverseIoOwn is dropped OR when the IoContext is destroyed,
+// whichever comes first. Accessing the ReverseIoOwn after the IoContext is destroyed will throw.
```

Comment:
This could use a bit more detail on when/why someone would use a `ReverseIoOwn<T>` as opposed to an `IoOwn<T>` or whatnot. The `IoOwn<T>` on its own has already been a source of confusion for folks so I'd err on the side of over-documenting than under.